### ISSUE TITLE
[java] NPath complexity false negative due to overflow

### ISSUE THREAD

**Issue by maikelsteneker** [pmd/pmd#1226]
**Affects PMD Version:** 6.5.0

**Rule:** net.sourceforge.pmd.lang.java.rule.design.NPathComplexityRule

**Description:** On some methods, the NPath complexity is not reported, even though the threshold for reporting it is set to the minimum value. In earlier versions of PMD (e.g., 5.3), NPath complexity was reported for these methods.

**Code Sample demonstrating the issue:**

No NPath complexity is reported for the method `readVar()` in the sample below:

[XMLConfigSerializer.java.txt](https://github.com/pmd/pmd/files/2238036/XMLConfigSerializer.java.txt)

**Running PMD through:** CLI

The ruleset: [pmd.xml.gz](https://github.com/pmd/pmd/files/2162426/pmd.xml.gz)

```
    <rule ref="category/java/design.xml/NPathComplexity">
        <properties>
            <property name="reportLevel" value="0"/>
        </properties>
    </rule>
```

**Comment by oowekyala** [comment]
@jsotuyod I don't think that would interact well with the metrics' caching since every rule may have a separate report level. Granted, for now metrics are mostly one-to-one with rules, so the caching may not be very useful (until we implement multifile analysis). 

For this metric we could probably just use a `long`, or a BigInteger, and clip the computed value to Double.MAX_VALUE before caching

**Comment by adangel** [comment]
With PMD 5.3.8:
`/home/andreas/PMD/source/pmd-it/issue-1226/src/XMLConfigSerializer.java:263:	The method readVar() has an NPath complexity of 613528536`
